Client.Core/Flux/Internal/FluxResultMapper.cs

```diff
@@ -92,6 +92,76 @@ internal T ToPoco<T>(FluxRecord record)
             }
         }
 
+        /// <summary>
+        /// Maps FluxRecord into custom POCO class.
+        /// </summary>
+        /// <param name="record">the Flux record</param>
+        /// <param name="type">the POCO type</param>
+        /// <returns></returns>
+        /// <exception cref="InfluxException"></exception>
+        internal object ToPoco(FluxRecord record, Type type)
+        {
+            Arguments.CheckNotNull(record, "Record is required");
+
+            try
+            {
+                var poco = Activator.CreateInstance(type);
+
+                // copy record to case insensitive dictionary (do this once)
+                var recordValues =
+                    new Dictionary<string, object>(record.Values, StringComparer.InvariantCultureIgnoreCase);
+
+                var properties = PropertyCache.GetOrAdd(type, _ => type.GetProperties());
+
+                foreach (var property in properties)
+                {
+                    var attribute = AttributeCache.GetOrAdd(property, _ =>
+                    {
+                        var attributes = property.GetCustomAttributes(typeof(Column), false);
+                        return attributes.Length > 0 ? attributes[0] as Column : null;
+                    });
+
+                    if (attribute != null && attribute.IsTimestamp)
+                    {
+                        SetFieldValue(poco, property, record.GetTime());
+                    }
+                    else
+                    {
+                        var columnName = property.Name;
+
+                        if (attribute != null && !string.IsNullOrEmpty(attribute.Name))
+                        {
+                            columnName = attribute.Name;
+                        }
+
+                        string col = null;
+
+                        if (recordValues.ContainsKey(columnName))
+                        {
+                            col = columnName;
+                        }
+                        else if (recordValues.ContainsKey("_" + columnName))
+                        {
+                            col = "_" + columnName;
+                        }
+
+                        if (!string.IsNullOrEmpty(col))
+                        {
+                            // No need to set field value when column does not exist (default poco field value will be the same)
+                            if (recordValues.TryGetValue(col, out var value))
+                                SetFieldValue(poco, property, value);
+                        }
+                    }
+                }
+
+                return poco;
+            }
+            catch (Exception e)
+            {
+                throw new InfluxException(e);
+            }
```

Comment:
This code is duplicated from: `internal T ToPoco<T>(FluxRecord record)`. Please try to reuse code to support better maintenance.